components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CorrelationConfigDAO.java

```diff
@@ -69,7 +74,12 @@ public boolean updateCorrelationConfigs(List<CorrelationConfigDTO> correlationCo
             log.debug("Updating Correlation Configs");
             try (PreparedStatement preparedStatementConfigs = connection.prepareStatement(queryConfigs)) {
                 for (CorrelationConfigDTO correlationConfigDTO : correlationConfigDTOList) {
-                    String componentName = correlationConfigDTO.getName();
+                    String componentName = correlationConfigDTO.getName().trim();
+                    if (!correlationComponents.contains(componentName)) {
```

Comment:
Can't we check component validity at `DevopsAPIUtils.validateCorrelationComponentList()`?

Reply:
In `DevopsAPIUtils.validateCorrelationComponentList()`, we do not have access to the correlation component list provided in the api-manager.xml

Reply:
Shall we check the validity before creating the SQL connections and statements? Else it's just a waste of resources. @tharindu1st WDYT?